python/lsst/cell_coadds/_fits.py

```diff
@@ -0,0 +1,407 @@
+        """Read the FITS file as a StitchedCoadd object."""
+        coadd = self.readAsMultipleCellCoadd()
+        with fits.open(self.filename) as hdu_list:
+            header = hdu_list[1].header
+        minY = header["STBBOX12"]
+        minX = header["STBBOX11"]
+        maxX = header["STBBOX21"]
+        maxY = header["STBBOX22"]
+        bbox = Box2I(Point2I(minX, minY), Point2I(maxX, maxY))
+        return coadd.stitch(bbox=bbox)
+
+    @staticmethod
+    def _readSingleCellCoadd(
+        data: Mapping[str, Any], header: Mapping[str, Any], common: CommonComponents
+    ) -> SingleCellCoadd:
+        """Read a coadd from a FITS file.
+
+        Parameters
+        ----------
+        data : `Mapping`
+            The data from the FITS file.
+        header : `dict`
+            The header of the FITS file.
+        common : `CommonComponents`
+            The common components of the coadd.
+
+        Returns
+        -------
+        coadd : `SingleCellCoadd`
+            The coadd read from the file.
+        """
+        buffer_x, buffer_y = (header["OCELL1"] - header["GRCELL1"]) // 2, (
+            header["OCELL2"] - header["GRCELL2"]
+        ) // 2
+        psf = ImageD(
+            array=data["psf"].astype(np.float64),
+            xy0=Point2I(-(header["PSFSIZE1"] // 2), -(header["PSFSIZE2"] // 2)),
+        )  # use the variable
+        xy0 = Point2I(
+            header["GRCELL1"] * data["cell_id"][0] - buffer_x,
+            header["GRCELL2"] * data["cell_id"][1] - buffer_y,
+        )
+        mask = afwImage.Mask(data["mask"].astype(np.int32), xy0=xy0)
+        image_planes = OwnedImagePlanes(
+            image=ImageF(
+                data["image"].astype(np.float32),
+                xy0=xy0,
+            ),
+            mask=mask,
+            variance=ImageF(data["variance"].astype(np.float32), xy0=xy0),
+            noise_realizations=[],
+            mask_fractions={},
+        )
+
+        identifiers = CellIdentifiers(
+            cell=Index2D(data["cell_id"][0], data["cell_id"][1]),
+            skymap=common.identifiers.skymap,
+            tract=common.identifiers.tract,
+            patch=common.identifiers.patch,
+            band=common.identifiers.band,
+        )
+
+        return SingleCellCoadd(
+            outer=image_planes,
+            psf=psf,
+            inner_bbox=Box2I(
+                minimum=Point2I(
+                    header["GRCELL1"] * data["cell_id"][0],
+                    header["GRCELL2"] * data["cell_id"][1],
+                ),
+                maximum=Point2I(
+                    header["GRCELL1"] * (data["cell_id"][0] + 1) - 1,
+                    header["GRCELL2"] * (data["cell_id"][1] + 1) - 1,
+                ),
+            ),
+            common=common,
+            identifiers=identifiers,
+            inputs=None,  # type: ignore[arg-type]  # TODO: Pass a sensible value here.
+        )
+
+    # TODO: Make this reader a context manager that handles file closures.
```

Comment:
These ticket-less TODOs will be fixed before we migrate to `lsst` organization on Github.